libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/location/ConstantVelocityInterpolator.kt

```diff
@@ -45,7 +39,23 @@ class ConstantVelocityInterpolator(
     override fun getInterpolation(input: Float): Float =
         innerInterpolator.getInterpolation(input)
 
-    private fun distanceTo(p1: Point, p2: Point): Double {
+    private fun distance(p1: Point, p2: Point): Double {
         return hypot(p2.latitude() - p1.latitude(), p2.longitude() - p1.longitude())
     }
+
+    private fun timingPath(distances: List<Double>, total: Double): Path {
+        val path = Path()
+        val step = 1.0f / distances.size
+        var pathTime = 0.0f
+        // NOTE: The Path must start at (0,0) and end at (1,1)
```

Comment:
What's your guess? Some precision loss when working with floats and doubles?
Because I was investigating this too and did not find any reason why it should not become (1f, 1f) naturally.
Are you sure it's a good idea to fix the bug blindly? I would try to ask the customer which input data caused the crash and try to look for the problem. Or try reproduce it with random input data (I did not succeed but I didn't try too hard). 
Maybe there is a hidden bug we don't see and this solution will just hide it deeper. Or maybe it will even create a weird unexpected path (because for some reason we don't get (1f, 1f) naturally).

Reply:
> What's your guess? Some precision loss when working with floats and doubles?
Yup. That's precisely what I am suspecting. Just the fact that only ~0.05% of customer users are affected makes me think there is some floating point "spookiness" at play here.

> Are you sure it's a good idea to fix the bug blindly? I would try to ask the customer which input data caused the crash and try to look for the problem. Or try reproduce it with random input data (I did not succeed but I didn't try too hard).

I think this solution is much safer than the previous one. PathInterpolator only accepts a Path that starts at (0,0) and ends at (1,1). By explicitly setting path endpoints and calculating points in between, we'll ensure that the Path will always be correct.

> Maybe there is a hidden bug we don't see and this solution will just hide it deeper. Or maybe it will even create a weird unexpected path (because for some reason we don't get (1f, 1f) naturally).

I don't think this will hide it deeper. Without the exact set of key points, devices that crashed occurred on and reliable repro septs we will never be able to repro this. 
Generating a weird, unexpected path that still can be used to instantiate PathInterpolator, will make this issue VISUAL which is far better than CRASH.

Reply:
Ideally we would send debug info to some analytics if we don't get (1f, 1f) naturally and force it to become (1f, 1f). But I don't think we have means of doing that, do we?

> Generating a weird, unexpected path that still can be used to instantiate PathInterpolator, will make this issue VISUAL which is far better than CRASH.

Well, yes, but still I'm not sure that fixing the crash is worth the risk of introducing weird and unexpected behaviour. Here I guess we have a valid hypothesis and it looks like the only reasonable way to fix this. OK then...

Reply:
> Ideally we would send debug info to some analytics if we don't get (1f, 1f) naturally and force it to become (1f, 1f). But I don't think we have means of doing that, do we?

Not that I am aware of. Good idea for Q1 project.